Warmstart documentation (#3332)
LBNL-UCB-STI/beam: docs/users.rst

* Environment variables: PWD=/path/to/beam/folder

11. Warmstart

> We can configure beam to run from previous state. This configuration can be controlled by following parameters in config file::

Reviewer: ```suggestion
Beam can be configured to run from a previous, stored state - referred to as warm start. A warm start can be controlled through the following config file parameters::
```

Reviewer: Minor grammatical tweaks - take it or leave it

Author: Applied Suggestion

 * beam.warmStart.type = "disabled"
 * beam.warmStart.path = "https://s3.us-east-2.amazonaws.com/beam-outputs/run140-base__2018-06-26_22-20-49_28e81b6d.zip"
 * beam.warmStart.samplePopulationIntegerFlag = 0
 * beam.warmStart.prepareData = false
 * beam.warmStart.skimsFilePaths = [
                                     {
                                         skimType = "String"
                                         skimsFilePath = ""
                                     }
                                   ]

> We can configure each param as following

Reviewer: ```suggestion
Each parameter is defined below::
```

Author: applied suggestion

> - `beam.warmStart.type`: This parameter control type of warm start. We have following options for this param.

Reviewer: ```suggestion
- `beam.warmStart.type`: This parameter controls the type of warm start, with the possibilities being::
```

Reviewer: Does the other colon need to be doubled?

Author: Applied suggestion. But not double colon. double colon have special meaning in Markdown language . Its create a block .

 - **disabled** - completely disable warm start

>  - **full** - All warm start files has been used for initialising warm start

Reviewer: ```suggestion
 - **full** - load all the data from the supplied files for initializing warm start
```

Author: applied suggestion

>  - **linkStatsOnly** - only link stats is loaded (all the other data is got from the input directory)

Reviewer: ```suggestion
 - **linkStatsOnly** - only load link stats data with the rest of the data retrieved from the input directory
```

Author: applied suggestion

>  - **linkStatsFromLastRun** - only link stats is loaded from beam.input.lastBaseOutputDir directory

Reviewer: ```suggestion
 - **linkStatsFromLastRun** - only load link stats from the `beam.input.lastBaseOutputDir` directory
```

Reviewer: Is this like a full, but with the link stats loaded from last run? or is it ONLY link stats - being loaded from the last run?

Author: its only link stats are loaded from last run

> - `beam.warmStart.path`: This property consist path of warm starts files from some previous run. We can give path of zip file on s3 or local dir

Reviewer: ```suggestion
- `beam.warmStart.path`: This parameter is the path where warm start files from some previous run can be found. It can be the path of a zip file on s3 or local directory
```

Author: applied suggestion

> - `beam.warmStart.samplePopulationIntegerFlag`: This parameter control if population sampling needed or not. For full warmstart run we can not change sampling

Reviewer: ```suggestion
- `beam.warmStart.samplePopulationIntegerFlag`: This parameter controls if population sampling is needed or not. For full warmstart this will be ignored, as sampling cannot be changed.
```

Author: applied suggestion

> - `beam.warmStart.prepareData`: This parameter control if we have to prepare warmstart archive for distribution. If true then beam write following file in archive.

Reviewer: ```suggestion
- `beam.warmStart.prepareData`: This parameter controls if warmstart need to prepare an archive for distribution. If true then beam writes the following file in the archive.
```

Author: applied suggestion

>    Root level files

Reviewer: Can you provide more details on what these are? Is it the data that will be archived?

Author: yes this is the data for archive for warmstart

Reviewer: Can this be added to the docs, though?

      output_personAttributes.xml.gz, population.csv.gz,households.csv.gz,vehicles.csv.gz

   Iteration level files

      linkstats.csv.gz,plans.csv.gz,plans.xml.gz,rideHailFleet.csv.gz

> - `beam.warmStart.skimsFilePaths`: This param is used for configuring type of skim and path of skim file

Reviewer: ```suggestion
- `beam.warmStart.skimsFilePaths`: This parameter is used for configuring the type of skim and path of the skim file
```

Reviewer: How does it do both?

Author: this parameter have 2 more parameters to configure path and type of skim. added this details in doc
